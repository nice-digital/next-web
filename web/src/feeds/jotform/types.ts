export type FormID = `${number}`;

export type Duration = `${number}ms`;

export type NotFoundResponse = {
	responseCode: 404;
	message: string;
	content: "";
	duration: Duration;
	info: string;
};

export type NotAuthorizedResponse = {
	responseCode: 401;
	message: string;
	content: "";
	duration: Duration;
	info: string;
};

export type FormStatus = "ENABLED" | "DISABLED";

export type FormType = "LEGACY" | "CARD";

export type FormSuccessResponse = {
	responseCode: 200;
	message: "success";
	content: {
		id: string;
		username: string;
		title: string;
		height: number;
		status: FormStatus;
		/** Format like "2023-03-21 09:27:18" */
		created_at: string;
		updated_at: string;
		last_submission: string;
		new: `${number}`;
		count: `${number}`;
		type: FormType;
		favorite: `${number}`;
		archived: `${number}`;
		url: string;
	};
	duration: Duration;
};

export type GetFormResponse =
	| NotAuthorizedResponse
	| NotFoundResponse
	| FormSuccessResponse;
