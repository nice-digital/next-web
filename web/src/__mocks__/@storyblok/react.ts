export const storyblokInit = (): null => null;

export const apiPlugin = (): null => null;
