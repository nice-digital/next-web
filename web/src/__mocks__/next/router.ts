export const useRouter = jest.fn().mockImplementation(() => {
	throw new Error("Provide a mock of useRouter for testing purposes");
});
