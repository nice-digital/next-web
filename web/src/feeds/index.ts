import { type Readable } from "stream";

import axios, { type AxiosInstance } from "axios";
import applyCaseMiddleware from "axios-case-converter";
import { camelCase } from "camel-case";

import { cache, getCacheKey } from "@/cache";

export const client: AxiosInstance = applyCaseMiddleware(
	axios.create({
		headers: { accept: "application/json" },
	}),
	{
		caseFunctions: {
			camel: (input, _options) => {
				// Strip out the prefix from all the embedded keys as it creates so much noise
				return camelCase(input.replace("nice.publications:", ""));
			},
		},
	}
);

/**
 * Gets the body of a feed directly from the back end system
 *
 * @param path The path of the feed to request
 * @returns The body of the feed
 */
export const getFeedBodyUnCached = async <TResponse>(
	origin: string,
	path: string,
	apiKey: string,
	acceptHeader = "application/json"
): Promise<TResponse> => {
	const { data } = await client.get<TResponse>(origin + path, {
		headers: {
			"Api-Key": apiKey,
			Accept: acceptHeader,
		},
		validateStatus: (status: number) => {
			// We don't want feed 404 responses to throw an error, so that we can show users a not found page rather than a server error.
			return (status >= 200 && status < 300) || status == 404;
		},
	});

	return data;
};

/**
 * Gets the body of a feed from the cache, if it exists.
 * If the item isn't in cache then it will call the `getUncachedAction` function to get fresh data.
 *
 * @param path The path of the feed endpoint to request
 * @param ttl The TTL (time to live) of the entry
 * @param getUncachedAction Function that gets the uncached feed directly from the back end.
 * @returns The body of the feed
 */
export const getFeedBodyCached = async <T>(
	groupCacheKey: string,
	path: string,
	ttl: number,
	getUncachedAction: () => Promise<T>
): Promise<T> =>
	cache.wrap<T>(getCacheKey(groupCacheKey.toLowerCase(), path.toLowerCase()), getUncachedAction, { ttl });

/**
 * Gets a response stream from a remote API endpoint, usually used for binary files e.g. PDFs (or mobi/epub etc).
 *
 * @param origin The base URL of the endpoint
 * @param path The path to the endpoint e.g. `/feeds/downloads/737585a0-dad7-4a37-875b-b30b09c3fdc3`
 * @param apiKey The API key for authentication against the API
 * @returns A readable response stream
 */
export const getResponseStream = async (
	origin: string,
	path: string,
	apiKey: string
): Promise<Readable> => {
	const { data } = await axios.get<Readable>(origin + path, {
		headers: {
			"Api-Key": apiKey,
		},
		responseType: "stream",
		maxBodyLength: Infinity,
	});
	return data;
};
