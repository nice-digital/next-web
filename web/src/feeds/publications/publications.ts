import { serverRuntimeConfig } from "@/config";

import { getFeedBodyCached, getFeedBodyUnCached, getResponseStream } from "../";

import {
	AreaOfInterest,
	AreaOfInterestTypes,
	ErrorResponse,
	FeedPath,
	ChapterHTMLContent,
	IndicatorSubType,
	IndicatorSubTypes,
	ProductDetail,
	ProductListLite,
	ProductLite,
	ProductType,
	ProductTypes,
	ResourceDetail,
	RelatedResourceList,
	IndicatorMappings,
	IndicatorMapping,
} from "./types";

export * from "./types";

const cacheKeyPrefix = "publications",
	{ defaultTTL, longTTL } = serverRuntimeConfig.cache,
	{ origin, apiKey } = serverRuntimeConfig.feeds.publications;

/**
 * Gets a list of products from the 'products lite' endpoint.
 */
export const getAllProducts = async (): Promise<ProductLite[]> =>
	await getFeedBodyCached<ProductLite[]>(
		cacheKeyPrefix,
		FeedPath.ProductsLite,
		defaultTTL,
		async () =>
			(
				await getFeedBodyUnCached<ProductListLite>(
					origin,
					FeedPath.ProductsLite,
					apiKey
				)
			).embedded.productListLite.embedded.productLite.map((product) => {
				// Discard unneeded properties on products to make what we're storing in cache a lot smaller.
				// This means we're essentially storing a ProductLite in cache rather than a ProductLiteRaw.
				// In perf tests this saved ~30% off the cache load time from the file system (once you factor in deserialization, file access times etc).
				delete (product as Partial<typeof product>).eTag;
				delete (product as Partial<typeof product>).links;
				return product;
			})
	);

/**
 * Gets _all_ product types.
 *
 * Note: there's no pre-filter so it includes both enabled _and_ disabled product types.
 */
export const getAllProductTypes = async (): Promise<ProductType[]> =>
	await getFeedBodyCached<ProductType[]>(
		cacheKeyPrefix,
		FeedPath.ProductTypes,
		longTTL,
		async () =>
			(
				await getFeedBodyUnCached<ProductTypes>(
					origin,
					FeedPath.ProductTypes,
					apiKey
				)
			).productTypes
	);

/**
 * Gets _all_ areas of interest e.g. Covid-19 and Antimicrbial prescribing.
 *
 * Note: there's no pre-filter so it includes both enabled _and_ disabled areas of interest.
 */
export const getAllAreasOfInterest = async (): Promise<AreaOfInterest[]> =>
	await getFeedBodyCached<AreaOfInterest[]>(
		cacheKeyPrefix,
		FeedPath.AreasOfInterest,
		longTTL,
		async () =>
			(
				await getFeedBodyUnCached<AreaOfInterestTypes>(
					origin,
					FeedPath.AreasOfInterest,
					apiKey
				)
			).areaOfInterestTypes
	);

/**
 * Gets _all_ indicator sub types e.g. Clinical commissioning group indicator.
 *
 * Note: there's no pre-filter so it includes both enabled _and_ disabled indicator sub types.
 */
export const getAllIndicatorSubTypes = async (): Promise<IndicatorSubType[]> =>
	await getFeedBodyCached<IndicatorSubType[]>(
		cacheKeyPrefix,
		FeedPath.IndicatorSubTypes,
		longTTL,
		async () =>
			(
				await getFeedBodyUnCached<IndicatorSubTypes>(
					origin,
					FeedPath.IndicatorSubTypes,
					apiKey
				)
			).indicatorSubTypes
	);

/**
 * Gets the indicator sub type object from the given identifier prefix
 *
 * @param identifierPrefix The identifier prefix of the indicator sub type to find e.g. CCG, GPIQ, GPINQ, NLQ etc
 * @returns The indicator sub type object if found, otherwise null
 */
export const getIndicatorSubType = async (
	identifierPrefix: string
): Promise<IndicatorSubType | null> =>
	(await getAllIndicatorSubTypes()).find(
		(subType) => subType.identifierPrefix === identifierPrefix
	) || null;

/**
 * Gets _all_ indicator mappings (redirects).
 */
export const getIndicatorMappings = async (): Promise<IndicatorMapping[]> =>
	await getFeedBodyCached<IndicatorMapping[]>(
		cacheKeyPrefix,
		FeedPath.IndicatorMappings,
		defaultTTL,
		async () =>
			(
				await getFeedBodyUnCached<IndicatorMappings>(
					origin,
					FeedPath.IndicatorMappings,
					apiKey
				)
			).mappings
	);

/**
 * Gets a full product detail response from publications from the given product id.
 *
 * @returns The full product details, or `null` if the product can't be found
 */
export const getProductDetail = async (
	productId: string
): Promise<ProductDetail | null> =>
	await getFeedBodyCached<ProductDetail | null>(
		cacheKeyPrefix,
		FeedPath.ProductDetail + productId,
		longTTL,
		async () => {
			const response = await getFeedBodyUnCached<ProductDetail | ErrorResponse>(
				origin,
				FeedPath.ProductDetail + productId,
				apiKey
			);

			return isSuccessResponse(response) ? response : null;
		}
	);

/**
 * Gets chapter HTML.
 *
 */
export const getChapterContent = async (
	chapterHref: string
): Promise<ChapterHTMLContent | null> => {
	return getFeedBodyCached<ChapterHTMLContent | null>(
		cacheKeyPrefix,
		chapterHref,
		longTTL,
		async () => {
			const response = await getFeedBodyUnCached<
				ChapterHTMLContent | ErrorResponse
			>(origin, chapterHref, apiKey);

			return isSuccessResponse(response) ? response : null;
		}
	);
};

/**
 * Fetches a full resource response from publications.
 *
 * Returns null if the resource can't be found.
 *
 * @param resource The related resource, or null if the resource can't be found
 */
export const getResourceDetail = async (
	resource: RelatedResourceList
): Promise<ResourceDetail | null> => {
	const { url } = resource;

	return getFeedBodyCached<ResourceDetail | null>(
		cacheKeyPrefix,
		url,
		longTTL,
		async () => {
			const response = await getFeedBodyUnCached<
				ResourceDetail | ErrorResponse
			>(origin, url, apiKey);

			return isSuccessResponse(response) ? response : null;
		}
	);
};

/**
 * Fetches full resources responses from publications for each of the given resources.
 *
 * It will throw an error if any of the resources can't be found
 */
export const getResourceDetails = async (
	resources: RelatedResourceList[]
): Promise<ResourceDetail[]> => {
	const fullResources = await Promise.all(
		resources.map(
			(relatedResource) =>
				new Promise<{
					relatedResource: RelatedResourceList;
					resourceDetail: ResourceDetail | null;
				}>((resolve) => {
					getResourceDetail(relatedResource).then((resourceDetail) => {
						resolve({
							relatedResource,
							resourceDetail,
						});
					});
				})
		)
	);

	// We can't 100% guarantee all related resources always still exist so handle if some can't be found
	const failedResources = fullResources
		.filter((r) => r.resourceDetail === null)
		.map((r) => r.relatedResource);

	if (failedResources.length)
		throw Error(
			`Could not load resources ${failedResources.map((r) => r.uid)}`
		);

	return fullResources
		.map((r) => r.resourceDetail)
		.filter((r): r is ResourceDetail => r !== null);
};

export const getImage = async (url: string): Promise<Buffer> =>
	await getFeedBodyCached<Buffer>(cacheKeyPrefix, url, longTTL, async () => {
		// Construct the full URL and headers for the request
		const logourl = origin + url;

		const response = await fetch(logourl, {
			method: "GET",
			headers: {
				"Api-Key": apiKey,
				Accept: "application/json",
			},
		});

		// Check if the response is OK
		if (!response.ok) {
			throw new Error(
				`Failed to fetch image from ${url}: ${response.statusText}`
			);
		}

		// Convert the response stream to a Blob
		const blob = await response.blob();

		let buffer = Buffer.from(await blob.arrayBuffer());

		return buffer;
	});

/**
 * Gets a stream of a file from publications.
 *
 * @param filePath The relative path of the endpoint that serves file content, e.g. `/feeds/downloads/737585a0-dad7-4a37-875b-b30b09c3fdc3`
 * @returns A readable stream of the file contents
 */
export const getFileStream = async (
	filePath: string
): Promise<ReturnType<typeof getResponseStream>> =>
	getResponseStream(origin, filePath, apiKey);

/**
 * A user-defined type guard for checking whether a given response is an error or not
 *
 * @param response The response
 * @returns True if this is an error response otherwise false
 */
export function isErrorResponse<TValidResponse>(
	response: TValidResponse | ErrorResponse
): response is ErrorResponse {
	return (response as ErrorResponse).statusCode !== undefined;
}

/**
 * A user-defined type guard for checking whether a given response is not an error.
 * The opposite of `isErrorResponse`
 *
 * @param response The response
 * @returns True if this is an error response otherwise false
 */
export function isSuccessResponse<TValidResponse>(
	response: TValidResponse | ErrorResponse
): response is TValidResponse {
	return (response as ErrorResponse).statusCode === undefined;
}
