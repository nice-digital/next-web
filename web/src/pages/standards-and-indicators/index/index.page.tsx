import { GetServerSideProps } from "next/types";

export const getServerSideProps: GetServerSideProps = async () => {
	return {
		redirect: {
			destination: "/indicators/published",
			permanent: true,
		},
	};
};

export default function OldIndicatorListPage(): void {
	// Default export to prevent next.js errors: we don't need an actual component
	// because the work is done in getServerSideProps to redirect
}
