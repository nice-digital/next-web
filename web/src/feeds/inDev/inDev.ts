import { serverRuntimeConfig } from "@/config";

import { getFeedBodyCached, getFeedBodyUnCached } from "..";
import { ErrorResponse } from "../publications/types";

import {
	Project,
	FeedPath,
	AllProjects,
	Consultation,
	InConsultationProjects,
	ProjectDetail,
	IndevPanel,
} from "./types";

export * from "./types";

const cacheKeyPrefix = "inDev",
	{ defaultTTL, longTTL } = serverRuntimeConfig.cache,
	{ origin, apiKey } = serverRuntimeConfig.feeds.inDev;

export const getAllProjects = async (): Promise<Project[]> =>
	await getFeedBodyCached<Project[]>(
		cacheKeyPrefix,
		FeedPath.AllProjects,
		defaultTTL,
		async () =>
			(
				await getFeedBodyUnCached<AllProjects>(
					origin,
					FeedPath.AllProjects,
					apiKey
				)
			).embedded.niceIndevIndevelopmentProject
	);

export const getAllConsultations = async (): Promise<Consultation[]> =>
	await getFeedBodyCached<Consultation[]>(
		cacheKeyPrefix,
		FeedPath.InConsultationProjects,
		defaultTTL,
		async () =>
			(
				await getFeedBodyUnCached<InConsultationProjects>(
					origin,
					FeedPath.InConsultationProjects,
					apiKey
				)
			)?.embedded?.niceIndevInconsultationProduct || []
	);

/**
 * Gets a project detail.
 *
 */

export const getProjectDetail = async (
	inDevReference: string
): Promise<ProjectDetail | ErrorResponse> =>
	//TODO don't cache error response
	await getFeedBodyCached<ProjectDetail | ErrorResponse>(
		cacheKeyPrefix,
		FeedPath.ProjectDetail + inDevReference,
		longTTL,
		async () =>
			await getFeedBodyUnCached<ProjectDetail | ErrorResponse>(
				origin,
				FeedPath.ProjectDetail + inDevReference,
				apiKey
			)
	);
