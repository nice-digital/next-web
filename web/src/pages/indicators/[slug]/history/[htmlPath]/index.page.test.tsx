import { render, screen } from "@testing-library/react";
import MockAdapter from "axios-mock-adapter";
import { type GetServerSidePropsContext } from "next";
import { useRouter } from "next/router";

import { FeedPath as IndevFeedPath } from "@/feeds/inDev/types";
import { client } from "@/feeds/index";
import { FeedPath } from "@/feeds/publications/types";
import mockProject from "@/mockData/inDev/feeds/projects/ProjectDetail.json";
import mockIndicatorSubTypes from "@/mockData/publications/feeds/products/indicator-sub-types.json";
import mockProductWithInDevReference from "@/mockData/publications/feeds/products/indicator-with-indev-reference.json";
import { arrayify } from "@/utils/array";
import { formatDateStr, stripTime } from "@/utils/datetime";

import HistoryHTMLPage, {
	getServerSideProps,
	HistoryHTMLPageProps,
} from "./index.page";

type HistoryHTMLPageGetServerSidePropsContext = GetServerSidePropsContext<{
	slug: string;
	htmlPath: string;
}>;

const axiosMock = new MockAdapter(client, {
	onNoMatch: "throwException",
});

describe("/indicators/[slug]/history.page", () => {
	const slug = "ind6-new-indicator-product-1";
	const htmlPath = "html-content";
	const mockEditableHTML = "<p>some test html content</p>";

	beforeEach(() => {
		(useRouter as jest.Mock).mockImplementation(() => ({
			asPath: `/indicators/${slug}/history/${htmlPath}`,
		}));
		axiosMock.reset();

		axiosMock
			.onGet(new RegExp(FeedPath.ProductDetail))
			.reply(200, mockProductWithInDevReference)
			.onGet(new RegExp(IndevFeedPath.ProjectDetail))
			.reply(200, mockProject)
			.onGet(new RegExp(FeedPath.IndicatorSubTypes))
			.reply(200, mockIndicatorSubTypes)
			.onGet(new RegExp(`/guidance/NG100/documents/${htmlPath}`))
			.reply(200, mockEditableHTML);

		jest.resetModules();
	});

	describe("HistoryPage", () => {
		let props: HistoryHTMLPageProps;
		beforeEach(async () => {
			const context = {
				params: { slug: slug, htmlPath: htmlPath },
			} as HistoryHTMLPageGetServerSidePropsContext;

			props = (
				(await getServerSideProps(context)) as {
					props: HistoryHTMLPageProps;
				}
			).props;
		});

		it("should match snapshot for main content", () => {
			render(<HistoryHTMLPage {...props} />);
			expect(document.body).toMatchSnapshot();
		});

		it("should render html content from the endpoint", () => {
			render(<HistoryHTMLPage {...props} />);
			expect(screen.getByText("some test html content")).toBeInTheDocument();
		});

		it.each([
			[
				"Comments form (Word, 65 kB)",
				"/indicators/ind6-new-indicator-product-1/history/downloads/IND6-comments-form.doc",
			],
			[
				"Draft scope (PDF, 124 kB)",
				"/indicators/ind6-new-indicator-product-1/history/downloads/IND6-draft-scope.pdf",
			],
			[
				"Equality impact assessment (PDF, 233 kB)",
				"/indicators/ind6-new-indicator-product-1/history/downloads/IND6-equality-impact-assessment-8.pdf",
			],
		])(
			"should render resource links for %s with correct %s link and file info",
			(linkText, href) => {
				render(<HistoryHTMLPage {...props} />);

				const resourceLink = screen.getByText(linkText);

				expect(resourceLink).toBeInTheDocument();
				expect(resourceLink).toHaveAttribute("href", href);
			}
		);

		it("should display the lastUpdated date", () => {
			render(<HistoryHTMLPage {...props} />);

			const resource = arrayify(
				mockProject._embedded["nice.indev:panel-list"]._embedded[
					"nice.indev:panel"
				][13]._embedded["nice.indev:resource-list"]._embedded[
					"nice.indev:resource"
				]
			);

			const resourcePublishedDate = resource[0].PublishedDate;

			expect(
				screen.getByText("This page was last updated on")
			).toBeInTheDocument();

			const time = screen.getByText(
				formatDateStr(resourcePublishedDate as string),
				{
					selector: "time",
				}
			);
			expect(time).toHaveAttribute(
				"dateTime",
				stripTime(resourcePublishedDate as string)
			);
		});
	});

	describe("getServerSideProps", () => {
		it("should return a correct props", async () => {
			const result = await getServerSideProps({
				params: { slug, htmlPath },
				resolvedUrl: `/indicators/${slug}/history/${htmlPath}`,
			} as HistoryHTMLPageGetServerSidePropsContext);

			expect(result).toMatchSnapshot();
		});

		it("should return a not found if resource title id doesn't match the htmlPath param", async () => {
			const wrongHtmlPath = "non-existent-html-1";

			const notFoundResult = await getServerSideProps({
				params: { slug, htmlPath: wrongHtmlPath },
				resolvedUrl: `/indicators/${slug}/history/${htmlPath}`,
			} as HistoryHTMLPageGetServerSidePropsContext);

			expect(notFoundResult).toStrictEqual({ notFound: true });
		});

		it("should return a not found if there are panels of type 'History'", async () => {
			const nonHistoryPanels = mockProject._embedded[
				"nice.indev:panel-list"
			]._embedded["nice.indev:panel"].filter(
				(panel) => panel.PanelType != "History"
			);

			axiosMock.onGet(new RegExp(IndevFeedPath.ProjectDetail)).reply(200, {
				...mockProject,
				_embedded: {
					"nice.indev:panel-list": {
						_embedded: {
							"nice.indev:panel": nonHistoryPanels,
						},
					},
				},
			});

			const notFoundResult = await getServerSideProps({
				params: { slug, htmlPath },
				resolvedUrl: `/indicators/${slug}/history/${htmlPath}`,
			} as HistoryHTMLPageGetServerSidePropsContext);

			expect(notFoundResult).toStrictEqual({ notFound: true });
		});

		it("should return not found if the resource html is null", async () => {
			axiosMock
				.onGet(new RegExp("/guidance/NG100/documents/html-content"))
				.reply(404, {
					Message: "Not found",
					StatusCode: "NotFound",
				});

			const notFoundResult = (await getServerSideProps({
				params: { slug, htmlPath },
				resolvedUrl: `/indicators/${slug}/history/${htmlPath}`,
			} as HistoryHTMLPageGetServerSidePropsContext)) as {
				props: HistoryHTMLPageProps;
			};

			expect(notFoundResult).toStrictEqual({ notFound: true });
		});
	});
});
