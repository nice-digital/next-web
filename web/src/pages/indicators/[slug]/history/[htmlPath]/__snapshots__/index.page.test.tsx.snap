// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/indicators/[slug]/history/[htmlPath]/index.page HistoryHTMLPage should match snapshot for main content 1`] = `
<body>
  <div>
    <nav
      aria-label="Breadcrumbs"
      data-component="breadcrumbs"
      role="navigation"
    >
      <ol
        class="breadcrumbs"
      >
        <li
          class="breadcrumbs__crumb"
        >
          <a
            href="/"
          >
            Home
          </a>
        </li>
        <li
          class="breadcrumbs__crumb"
        >
          <a
            href="/standards-and-indicators/indicators"
          >
            Indicators
          </a>
        </li>
        <li
          class="breadcrumbs__crumb"
        >
          <a
            href="/guidance/ng100"
          >
            NG100
          </a>
        </li>
        <li
          class="breadcrumbs__crumb"
        >
          <a
            href="/guidance/ng100/history"
          >
            History
          </a>
        </li>
        <li
          class="breadcrumbs__crumb"
        >
          <span>
            Draft scope consultation
          </span>
        </li>
      </ol>
    </nav>
    <script
      type="application/ld+json"
    >
      {
  "@context": "http://schema.org",
  "@type": "BreadcrumbList",
  "itemListElement": [
    {
      "@type": "ListItem",
      "position": 1,
      "item": {
        "@id": "/",
        "name": "Home"
      }
    },
    {
      "@type": "ListItem",
      "position": 2,
      "item": {
        "@id": "/standards-and-indicators/indicators",
        "name": "Indicators"
      }
    },
    {
      "@type": "ListItem",
      "position": 3,
      "item": {
        "@id": "/guidance/ng100",
        "name": "NG100"
      }
    },
    {
      "@type": "ListItem",
      "position": 4,
      "item": {
        "@id": "/guidance/ng100/history",
        "name": "History"
      }
    },
    {
      "@type": "ListItem",
      "position": 5,
      "item": {
        "name": "Draft scope consultation"
      }
    }
  ]
}
    </script>
    <div
      class="page-header "
      data-component="page-header"
      id="content-start"
    >
      <h1
        class="page-header__heading page-header__heading--alt"
      >
        Rheumatoid arthritis in adults: management
      </h1>
      <ul
        class="page-header__metadata"
      >
        <li>
          NICE guideline
        </li>
        <li>
          NG100
        </li>
        <li>
          Published:
          <time
            datetime="2018-07-11"
          >
             
            11 July 2018
          </time>
        </li>
        <li>
          Last updated:
          <time
            datetime="2020-10-12"
          >
             
             
            12 October 2020
          </time>
        </li>
      </ul>
    </div>
    <nav
      class="horizontal-nav"
    >
      <ul
        class="horizontal-nav__list"
      >
        <li
          class="horizontal-nav__item"
          data-component="horizontal-nav"
        >
          <a
            aria-current="false"
            class="horizontal-nav__link"
            href="/guidance/ng100"
          >
            Indicator
          </a>
        </li>
        <li
          class="horizontal-nav__item"
          data-component="horizontal-nav"
        >
          <a
            aria-current="false"
            class="horizontal-nav__link"
            href="/guidance/ng100/resources"
          >
            Tools and resources
          </a>
        </li>
        <li
          class="horizontal-nav__item"
          data-component="horizontal-nav"
        >
          <a
            aria-current="false"
            class="horizontal-nav__link"
            href="/guidance/ng100/evidence"
          >
            Evidence
          </a>
        </li>
        <li
          class="horizontal-nav__item"
          data-component="horizontal-nav"
        >
          <a
            aria-current="true"
            class="horizontal-nav__link"
            href="/guidance/ng100/history"
          >
            History
          </a>
        </li>
      </ul>
    </nav>
    <div>
      <p>
        You can now comment on this draft scope. The scope defines what the guideline will (and will not) cover.
      </p>
      

      <p>
        <strong>
          The consultation closes on 
        </strong>
        <strong>
          15 July 2016
        </strong>
         
        <strong>
          at 5pm
        </strong>
      </p>
      

      <h3>
        How to comment
      </h3>
      

      <h2>
        1. Register your organisation
      </h2>
      

      <p>
        Your organisation needs to be 
        <a
          href="https://www.nice.org.uk/get-involved/stakeholder-registration"
        >
          registered as a stakeholder
        </a>
         before we can accept comments.
      </p>
      

      <p>
        Not eligible? Contact the 
        <a
          href="https://www.nice.org.uk/guidance/GID-NG10014/documents/stakeholder-list"
        >
          stakeholder organisation
        </a>
         that most closely represents your interests and pass your comments to them.
      </p>
      

      <h2>
        2. Read the consultation documents
      </h2>
      

      <ul>
        

        <li>
          <a
            href="/guidance/GID-NG10014/documents/draft-scope"
          >
            Draft scope
          </a>
        </li>
        

      </ul>
      

      <p>
        <strong>
          Background documents
        </strong>
      </p>
      

      <ul>
        

        <li>
          <a
            href="/guidance/GID-NG10014/documents/equality-impact-assessment"
          >
            Equality impact assessment
          </a>
        </li>
        

        <li>
          <a
            href="http://www.nice.org.uk/article/pmg22/chapter/3-how-you-can-get-involved"
          >
            Developing NICE guidelines: how to get involved
          </a>
          <strong>
             
          </strong>
        </li>
        

      </ul>
      

      <h2>
        3. Make your comments
      </h2>
      

      <ul>
        

        <li>
          You must use the 
          <a
            href="/guidance/GID-NG10014/documents/comments-form"
          >
            Comments form
          </a>
        </li>
        

        <li>
          You must declare any links with, or funding from, the tobacco industry
        </li>
        

      </ul>
      

      <p>
        If you share similar views with another registered organisation, send a joint response.
      </p>
      

      <h2>
        4. Email the form to us
      </h2>
      

      <p>
        Email: 
        <a
          href="mailto:RAupdate@nice.org.uk"
        >
          RAupdate@nice.org.uk
        </a>
      </p>
      

      <p>
        <strong>
          Deadline:
        </strong>
        <strong>
          15 July 2016
        </strong>
        , 
        <strong>
          5pm
        </strong>
      </p>
    </div>
    <div
      class="resourceLinks"
    >
      <hr
        class="mb--d"
      />
      <ul
        class="list list--unstyled"
      >
        <li>
          <article
            class="card"
            data-component="card"
          >
            <header
              class="card__header"
            >
              <p
                class="card__heading"
              >
                <a
                  href="/guidance/ng100/history/downloads/NG100-comments-form.doc"
                >
                  Comments form
                   
                  (
                  Word
                  ,
                   
                  65 kB
                  )
                </a>
              </p>
            </header>
            <dl
              class="card__metadata"
            >
              <div
                class="card__metadatum"
              >
                <dt
                  class="visually-hidden"
                >
                  Date
                </dt>
                <dd>
                  <time
                    datetime="2016-06-14"
                  >
                    14 June 2016
                  </time>
                </dd>
              </div>
            </dl>
          </article>
        </li>
        <li>
          <article
            class="card"
            data-component="card"
          >
            <header
              class="card__header"
            >
              <p
                class="card__heading"
              >
                <a
                  href="/guidance/ng100/history/downloads/NG100-draft-scope.pdf"
                >
                  Draft scope
                   
                  (
                  PDF
                  ,
                   
                  124 kB
                  )
                </a>
              </p>
            </header>
            <dl
              class="card__metadata"
            >
              <div
                class="card__metadatum"
              >
                <dt
                  class="visually-hidden"
                >
                  Date
                </dt>
                <dd>
                  <time
                    datetime="2016-06-17"
                  >
                    17 June 2016
                  </time>
                </dd>
              </div>
            </dl>
          </article>
        </li>
        <li>
          <article
            class="card"
            data-component="card"
          >
            <header
              class="card__header"
            >
              <p
                class="card__heading"
              >
                <a
                  href="/guidance/ng100/history/downloads/NG100-equality-impact-assessment-8.pdf"
                >
                  Equality impact assessment
                   
                  (
                  PDF
                  ,
                   
                  233 kB
                  )
                </a>
              </p>
            </header>
            <dl
              class="card__metadata"
            >
              <div
                class="card__metadatum"
              >
                <dt
                  class="visually-hidden"
                >
                  Date
                </dt>
                <dd>
                  <time
                    datetime="2020-03-31"
                  >
                    31 March 2020
                  </time>
                </dd>
              </div>
            </dl>
          </article>
        </li>
      </ul>
    </div>
    <p>
      This page was last updated on
       
      <time
        datetime="2016-07-19"
      >
        19 July 2016
      </time>
    </p>
  </div>
</body>
`;

exports[`/indicators/[slug]/history/[htmlPath]/index.page getServerSideProps should return a correct props 1`] = `
Object {
  "props": Object {
    "hasEvidenceResources": true,
    "hasHistory": true,
    "hasInfoForPublicResources": true,
    "hasToolsAndResources": true,
    "lastUpdated": "2016-07-19T08:56:00",
    "product": Object {
      "id": "NG100",
      "lastMajorModificationDate": "2020-10-12T00:00:00",
      "productTypeName": "NICE guideline",
      "publishedDate": "2018-07-11T00:00:00",
      "title": "Rheumatoid arthritis in adults: management",
    },
    "productPath": "/guidance/ng100",
    "resource": Object {
      "resourceFileHTML": "<p>You can now comment on this draft scope. The scope defines what the guideline will (and will not) cover.</p>
<p><strong>The consultation closes on </strong><strong>15 July 2016</strong> <strong>at 5pm</strong></p>
<h3>How to comment</h3>
<h2>1. Register your organisation</h2>
<p>Your organisation needs to be <a href=\\"https://www.nice.org.uk/get-involved/stakeholder-registration\\">registered as a stakeholder</a> before we can accept comments.</p>
<p>Not eligible? Contact the <a href=\\"https://www.nice.org.uk/guidance/GID-NG10014/documents/stakeholder-list\\">stakeholder organisation</a> that most closely represents your interests and pass your comments to them.</p>
<h2>2. Read the consultation documents</h2>
<ul>
<li><a href=\\"/guidance/GID-NG10014/documents/draft-scope\\">Draft scope</a></li>
</ul>
<p><strong>Background documents</strong></p>
<ul>
<li><a href=\\"/guidance/GID-NG10014/documents/equality-impact-assessment\\">Equality impact assessment</a></li>
<li><a href=\\"http://www.nice.org.uk/article/pmg22/chapter/3-how-you-can-get-involved\\">Developing NICE guidelines: how to get involved</a><strong>&nbsp;</strong></li>
</ul>
<h2>3. Make your comments</h2>
<ul>
<li>You must use the <a href=\\"/guidance/GID-NG10014/documents/comments-form\\">Comments form</a></li>
<li>You must declare any links with, or funding from, the tobacco industry</li>
</ul>
<p>If you share similar views with another registered organisation, send a joint response.</p>
<h2>4. Email the form to us</h2>
<p>Email: <a href=\\"mailto:RAupdate@nice.org.uk\\">RAupdate@nice.org.uk</a></p>
<p><strong>Deadline:</strong><strong>15 July 2016</strong>, <strong>5pm</strong></p>",
      "title": "Draft scope consultation",
    },
    "resourceLinks": Array [
      Object {
        "date": "2016-06-14T08:06:00",
        "fileSize": 64512,
        "fileTypeName": "Word",
        "href": "/guidance/ng100/history/downloads/NG100-comments-form.doc",
        "title": "Comments form",
        "type": "Draft scope consultation",
      },
      Object {
        "date": "2016-06-17T01:01:00",
        "fileSize": 123626,
        "fileTypeName": "PDF",
        "href": "/guidance/ng100/history/downloads/NG100-draft-scope.pdf",
        "title": "Draft scope",
        "type": "Draft scope consultation",
      },
      Object {
        "date": "2020-03-31T16:01:00",
        "fileSize": 233482,
        "fileTypeName": "PDF",
        "href": "/guidance/ng100/history/downloads/NG100-equality-impact-assessment-8.pdf",
        "title": "Equality impact assessment",
        "type": "Draft scope consultation",
      },
    ],
  },
}
`;
